seed/seed20.go

```diff
@@ -596,10 +594,65 @@ func (s *seed20) lookupUnassertedComponent(comp20 internal.Component20, info *sn
 	}
 	return Component{
 		Path:         compPath,
-		CompSideInfo: csi,
+		CompSideInfo: *csi,
 	}, nil
 }
 
+func (s *seed20) deriveSideInfo(snapRef naming.SnapRef, modelSnap *asserts.ModelSnap, optSnap *internal.Snap20, handler ContainerHandler, snapsDir string, tm timings.Measurer) (path string, sideInfo *snap.SideInfo, seedComps []Component, err error) {
+	var snapRev *asserts.SnapRevision
+	var snapDecl *asserts.SnapDeclaration
+	path, snapRev, snapDecl, err = s.lookupVerifiedRevision(snapRef, handler, snapsDir, tm)
+	if err != nil {
+		return
+	}
+	sideInfo = snapasserts.SideInfoFromSnapAssertions(snapDecl, snapRev)
+
+	if modelSnap != nil {
+		seedComps = make([]Component, 0, len(modelSnap.Components))
+		for comp, modelComp := range modelSnap.Components {
+			var seedComp Component
```

Comment:
Same here

Reply:
... but these ones I agree it is good to remove